addons/website_event_track/models/event_event.py

```diff
@@ -103,38 +92,66 @@ def toggle_website_track(self, val):
-                    menu = super(Event, event)._create_menu(sequence, name, url, xml_id)
-                    event.env['website.event.menu'].create({
-                        'menu_id': menu.id,
-                        'event_id': event.id,
-                        'menu_type': menu_type,
-                    })
-            elif not event.website_track:
-                event.track_menu_ids.mapped('menu_id').unlink()
-            if event.website_track_proposal and not event.track_proposal_menu_ids:
-                for sequence, (name, url, xml_id, menu_type) in enumerate(event._get_track_proposal_menu_entries()):
-                    menu = super(Event, event)._create_menu(sequence, name, url, xml_id)
-                    event.env['website.event.menu'].create({
-                        'menu_id': menu.id,
-                        'event_id': event.id,
-                        'menu_type': menu_type,
-                    })
-            elif not event.website_track_proposal:
-                event.track_proposal_menu_ids.mapped('menu_id').unlink()
+            if not menus_update_by_field or event in menus_update_by_field.get('website_track'):
+                event._update_website_menu_entry('website_track', 'track_menu_ids', '_get_track_menu_entries')
+            if not menus_update_by_field or event in menus_update_by_field.get('website_track_proposal'):
+                event._update_website_menu_entry('website_track_proposal', 'track_proposal_menu_ids', '_get_track_proposal_menu_entries')
+
+    def _update_website_menu_entry(self, fname_bool, fname_o2m, method_name):
+        self.ensure_one()
+        new_menu = None
+
+        if self[fname_bool] and not self[fname_o2m]:
+            for sequence, menu_data in enumerate(getattr(self, method_name)()):
+                if len(menu_data) == 4:
+                    (name, url, xml_id, menu_type) = menu_data
+                    menu_sequence = sequence
+                elif len(menu_data) == 5:
+                    (name, url, xml_id, menu_sequence, menu_type) = menu_data
+                new_menu = self._create_menu(menu_sequence, name, url, xml_id, menu_type=menu_type)
+        elif not self[fname_bool]:
+            self[fname_o2m].mapped('menu_id').unlink()
+
+        return new_menu
+
+    def _create_menu(self, sequence, name, url, xml_id, menu_type=False):
+        """ Override menu creation from website_event to link a website.event.menu
+        to the newly create menu (either page and url). """
+        website_menu = super(Event, self)._create_menu(sequence, name, url, xml_id, menu_type=menu_type)
+        if menu_type:
+            self.env['website.event.menu'].create({
+                'menu_id': website_menu.id,
+                'event_id': self.id,
+                'menu_type': menu_type,
+            })
+        return website_menu
+
+    def _get_menu_type_field_matching(self):
+        return {'track_proposal': 'website_track_proposal'}
```

Comment:
Not sure why you can't put 'track': 'website_track' here and avoir overhead in the unlink override?